docs/DeveloperGuide.adoc

```diff
@@ -262,6 +262,160 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
 
 Certain properties of the application can be controlled (e.g user prefs file location, logging level) through the configuration file (default: `config.json`).
 
+=== Book loaning feature
+
+==== Functionalities
+The functionalities and commands associated with the book loaning feauture are:
+
+* `loan sn/BOOK_SN`
+
+** Loans out a book based on its serial number.
+
+* `return INDEX [-all]`
+
+** Returns a book based on the index of that book shown on the GUI.
+
+* `renew INDEX [-all]`
+
+** Renews a book based on the index of that book shown on the GUI.
+
+* `reserve sn/BOOK_SN` or `reserve INDEX`
+
+** Reserves a book based on its serial number or index of that book shown on the GUI.
+
+* `pay AMOUNT`
+
+** Pay a fine amount for overdue books.
+
+The rationale for the different types of arguments for the loan, return, renew and reserve may not be apparent at first,
+but it is actually very simple.
+
+* When borrowers come to the librarian (our user) to borrow a book, the book itself would have
+a serial number. Thus, the user just types in the serial number of the book to be loaned out, instead of using find to
+locate that book in LiBerry to use its index.
+
+* Whereas, when a borrower comes to return a book, the librarian only sees a limited list of book that was loaned out by
+the borrower on the GUI. Hence, the user need not type the longer serial number to return that book, and instead, types
```

Comment:
maybe we can change the word `user` to `librarian` to avoid confusion between borrower and librarian